### PR TITLE
[6.3] org environment content access (Golden ticket) BZ:1498158, 1344049

### PR DESCRIPTION
**Pull request by ldjebran**
cover BZ:  https://bugzilla.redhat.com/show_bug.cgi?id=1498158
https://bugzilla.redhat.com/show_bug.cgi?id=1344049
depend on: https://github.com/SatelliteQE/nailgun/pull/465 (already merged)
```bash
pytest -v tests/foreman/api/test_contentaccess.py tests/foreman/cli/test_contentaccess.py tests/foreman/ui/test_contentaccess.py
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/2.7.13/envs/sat-6.3.0/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1, fauxfactory-1.0.1
collected 12 items                                                                                                      
2017-11-28 11:32:27 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_contentaccess.py::ContentAccessTestCase::test_positive_list_hosts_applicable <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_negative_rct_not_shows_golden_ticket_enabled <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_positive_erratum_installable <- robottelo/decorators/__init__.py FAILED
tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_positive_list_installable_updates <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_positive_rct_shows_golden_ticket_enabled <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_negative_list_available_packages <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_negative_visual_indicator_with_restricted_subscription <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_list_available_packages <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_list_installable_updates <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_activation_key_details <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_hosts_subscription <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_manifest <- robottelo/decorators/__init__.py PASSED

=================================== 1 failed, 8 passed, 3 skipped in 6909.22 seconds ===================================
```
```bash
pytest -v tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_positive_erratum_installable
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/2.7.13/envs/sat-6.3.0/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1, fauxfactory-1.0.1
collected 1 item                                                                                                        
2017-11-28 15:30:16 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/cli/test_contentaccess.py::ContentAccessTestCase::test_positive_erratum_installable <- robottelo/decorators/__init__.py PASSED

============================================== 1 passed in 257.01 seconds ==============================================

```

### CODE REVIEW

**Comment by ldjebran** [comment]
@oshtaier @rochacbruno @sghai Put do not merge for the moment as we need to investigate on some aspects with devs.

There is some related questions:
1. Is it OK to put rhel7 repo on tier2
2. I was a bit confused by "all installables should be available" in tests requirements , as per our procedures we install one package and verify that the needed installable/applicable are available.
What we are suggesting by that phrase: should we make a list of all rhel7 erratum and packages and verify (or one should sufficient)?

**Comment by rochacbruno** [comment]
@ldjebran for the **2** I think one is sufficient (my mistake when writing the stub requirement)

File: robottelo/manifests.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def _download_manifest_info(self):
             backend=default_backend()
         )
 
-    def clone(self):
+    def clone(self, org_environment_access=False):
```
Comment (oshtaier) [inline robottelo/manifests.py]: please update docstring to explain that new parameter and why we need it

File: robottelo/vm.py [hunk 2]
```diff
@@ -579,6 +579,25 @@ def configure_rhai_client(self, activation_key, org, rhel_distro):
                 'Unable to register client to Access Insights through '
                 'Satellite')
 
+    def patch_os_release_version(self, distro=None):
```
Comment (oshtaier) [inline robottelo/vm.py]: that method seems weird to me as anyway only one distro is supported, but even in that case, next code seems more appropriate:
```
+    def patch_os_release_version(self, distro=DISTRO_RHEL7):
 +        """Patch VM OS release version.
 +
 +        This is needed by yum package manager to generate the right RH
 +        repositories urls.
 +        """
 +        if distro == DISTRO_RHEL7:
```

Comment (ldjebran) [reply robottelo/vm.py]: @oshtaier we will add an other distro very soon to install virt-who server on rhel6 :)
and we are using this patch in many places

**Comment by ldjebran** [comment]
```console
pytest -v tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase
============================================ test session starts =============================================
platform linux2 -- Python 2.7.14, pytest-3.2.3, py-1.5.2, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/2.7.14/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1, fauxfactory-1.0.1
collected 7 items                                                                                             
2017-12-07 17:37:14 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_negative_list_available_packages <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_negative_visual_indicator_with_restricted_subscription <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_list_available_packages <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_list_installable_updates <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_activation_key_details <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_hosts_subscription <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentaccess.py::ContentAccessTestCase::test_positive_visual_indicator_on_manifest <- robottelo/decorators/__init__.py PASSED

=================================== 6 passed, 1 skipped in 2918.18 seconds ===================================
```

**Pull request merged**